src/phonehome.cpp

```diff
@@ -0,0 +1,98 @@
+#include "phonehome.h"
+#include <QDateTime>
+#include <QTime>
+
+using namespace adiscope;
+
+PHONEHOME_H
+
+PhoneHome::PhoneHome(QSettings *settings, Preferences *pref) :
+	ApiObject(), m_timestamp(0), m_versionsJson(""),
+	preferences(pref), settings(settings)
+{
+	setObjectName("phonehome");
+	load(*settings);
+}
+
+PhoneHome::~PhoneHome() {
+	save(*settings);
+}
+
+void PhoneHome::setPreferences(Preferences* preferences)
+{
+	PhoneHome::preferences = preferences;
+}
+
+void PhoneHome::versionsRequest(bool force)
+{
+	qint64 timeout = 24 * 60 * 60 * 1000; // 24 hours
+	//qint64 timeout = 5 * 1000;  // 5 seconds
+	qint64 now = QDateTime::currentMSecsSinceEpoch();
+
+	if(m_timestamp +  timeout < now  || force)
+	{
+		// from swdownloads
+		QNetworkAccessManager* manager = new QNetworkAccessManager();
```

Comment:
From my knowledge this types of network operation classes require a `EventLoop`, how this class gets it might be platform specific. I think it's better here to provide a parent (`this`), or even instantiate the `QNetworkAccessManager` with the `ToolLauncher` as parent.